File: projetDoki/src/main/java/formation/sopra/bu/projetDoki/metier/Adresse.java
package formation.sopra.bu.projetDoki.metier;

import javax.persistence.Column;
import javax.persistence.Embeddable;
import javax.persistence.OneToOne;
import javax.persistence.SequenceGenerator;

import com.fasterxml.jackson.annotation.JsonView;

import formation.sopra.bu.projetDoki.view.JsonViews;

@Embeddable
public class Adresse {
	@Column(name = "numero")
	@JsonView(JsonViews.Common.class)
	private Integer numero;
	
	@Column (name = "rue")
	@JsonView(JsonViews.Common.class)
	private String rue;
	
	@Column (name = "codePostal")
	@JsonView(JsonViews.Common.class)
	private String cp;
	
	@Column (name = "ville")
	@JsonView(JsonViews.Common.class)
	private String ville;
	
	@Column (name ="code")
	@JsonView(JsonViews.Common.class)
	private String code; 
	
	
	public Adresse() {}


	public Adresse(Integer numero, String rue, String cp, String ville, String code) {
		super();
		this.numero = numero;
		this.rue = rue;
		this.cp = cp;
		this.ville = ville;
		this.code = code;
	}
	

	public Integer getNumero() {
		return numero;
	}


	public void setNumero(Integer numero) {
		this.numero = numero;
	}


	public String getRue() {
		return rue;
	}


	public void setRue(String rue) {
		this.rue = rue;
	}


	public String getCp() {
		return cp;
	}


	public void setCp(String cp) {
		this.cp = cp;
	}


	public String getVille() {
		return ville;
	}


	public void setVille(String ville) {
		this.ville = ville;
	}


	public String getCode() {
		return code;
	}


	public void setCode(String code) {
		this.code = code;
	}
	
	
}

File: Doki/angular/src/app/model/adresse.ts
export class Adresse {

  private _numero: number;
  private _rue: string;
  private _cp: string;
  private _ville: string;
  private _code: string;


  constructor(numero?: number, rue?: string, cp?: string, ville?: string, code?: string) {
    this._numero = numero;
    this._rue = rue;
    this._cp = cp;
    this._ville = ville;
    this._code = code;
  }


  get numero(): number {
    return this._numero;
  }

  set numero(value: number) {
    this._numero = value;
  }

  get rue(): string {
    return this._rue;
  }

  set rue(value: string) {
    this._rue = value;
  }

  get cp(): string {
    return this._cp;
  }

  set cp(value: string) {
    this._cp = value;
  }

  get ville(): string {
    return this._ville;
  }

  set ville(value: string) {
    this._ville = value;
  }

  get code(): string {
    return this._code;
  }

  set code(value: string) {
    this._code = value;
  }
}
